FinHubSA/information-access
# chore: squashed a few bugs

# Pull Request

## Scope
https://dev.azure.com/JHNMAT015/ECO5037S%20Front%20end/_backlogs/backlog/ECO5037S%20Front%20end%20Team/Backlog%20items/?workitem=146
1. Name of download to reflect title. 
2. Page number to stay in center of results. 
3. Should be able to go backwards from preview without losing results.
4. Link provided in preview should be clickable. 
5. Author name and journal should be displayed. 

## Work Done
1. Inserted title as argument for download pdf function and now reflects title of paper. 
2. Added in a spacing div to keep the page number in middle of results. 
3. Use checkForSearch() method on mounting of search results to be able to go back from preview. 
4. Link on preview now has an anchor tag. 
5. Author name and journal placeholders were replaced with actual info. 

## Steps to Test
```npm run start```
1. Search for a paper, click the download button, and name of pdf should now be the title of the paper.
2. Search for something that has more than 2 results, change pages and page number shouldn't move around.
3. Search for a paper, go into preview. Now navigate backwards and it should not lose the previous search results. 
4. Search for a paper, go into preview, and click on link. 
5. Search for a paper, observe the card to see the info on it. 

## Screenshot
![image](https://user-images.githubusercontent.com/80746660/137103687-a10e38a5-8fd3-4a99-890d-f4a340a8e637.png)

![image](https://user-images.githubusercontent.com/80746660/137104236-57c7259e-220a-46e9-b1bb-a73136cdb45d.png)

![image](https://user-images.githubusercontent.com/80746660/137200196-d07d8dfc-1c8a-49e1-9dd5-67c328f6a3b8.png)
